assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -132,10 +147,54 @@ function FontPickerContainer({ value, onSelect, onClose }) {
   useFocusOut(ref, onClose, [onClose]);
 
   // Scroll to offset for current value
-  const currentOffset = fonts.findIndex(({ name }) => name === value);
+  const [currentOffset, setCurrentOffset] = useState(
+    fonts.findIndex(({ name }) => name === value)
+  );
 
-  // This is static for now, but with search and used fonts, this will change later
-  const matchingFonts = fonts;
+  const [matchingFonts, setMatchingFonts] = useState(fonts);
+
+  const [updateMatchingFonts] = useDebouncedCallback(
+    () => {
+      if (searchKeyword.trim() === '') {
+        setMatchingFonts(fonts);
+        return;
+      }
+      const _matchingFonts = fonts.filter(({ name }) =>
+        name.toLowerCase().includes(searchKeyword.toLowerCase())
+      );
+      setMatchingFonts(_matchingFonts);
+    },
+    250,
+    {},
+    [searchKeyword, fonts]
+  );
+
+  const handleSearchInputChanged = useCallback(
+    ({ target }) => {
+      setSearchKeyword(target.value);
+      updateMatchingFonts();
+    },
+    [updateMatchingFonts]
+  );
+
+  const handleInputFocus = useCallback(() => {
+    setCurrentOffset(-1);
```

Comment:
I don't feel that scroll properly resets once I start and stop searching.

Steps to reproduce scroll offset weirdness:

1. Add a text field and open the font picker
1. Select the last font in the list (_Yeon Sung_)
1. Open the font picker again, and observe that the list is scrolled to the bottom showing _Yeon Sung_ with a checkmark.
1. Focus the input field and type "r".
1. Observe the list still being scrolled to the very bottom, thus displayed a random selection of fonts matching "r" and not the first fonts matching.
1. Now clear the input field - and observe the scroll now being in a completely random spot in the list (matching the position of the last item of the search for "r", but completely irrelevant to the current view)

Reply:
Correct. This is implementing the search within the existing drop down. The offset functionality with auto load I think causes us some problems. We might have to rethink how this component works.